src/mapping/tests/RadialGeoMultiscaleMappingTest.cpp

```diff
@@ -120,5 +184,57 @@ BOOST_AUTO_TEST_CASE(testConsistentCollect)
   BOOST_TEST(outValues(2) == (inValues(4) + inValues(5)) / 2);
 }
 
+BOOST_AUTO_TEST_CASE(testConsistentCollectZ)
+{
+  /*  The following test works by creating two dimensionally heterogeneous meshes, namely 1D and 3D, that intersect along the x-axis.
+      Then, the data is mapped from the vertices of the 3D mesh to defined vertices of the 1D mesh (hence, COLLECT).
+      The defined vertices are close to a 1D vertex and therefore we can predict that the 1D vertex value should be the mean of all close 3D vertices.
+      Finally, this expected behavior is tested.  
+  */
+
+  PRECICE_TEST(1_rank);
+  int dimensions = 3;
+  using testing::equals;
+
+  // Create mesh to map from
+  PtrMesh inMesh(new Mesh("InMesh", dimensions, testing::nextMeshID()));
+  PtrData inData   = inMesh->createData("InData", 1, 0_dataID);
+  int     inDataID = inData->getID();
+  inMesh->createVertex(Eigen::Vector3d(0.0, 1.0, 1.0));
+  inMesh->createVertex(Eigen::Vector3d(0.0, 2.0, 1.0));
+  inMesh->createVertex(Eigen::Vector3d(0.0, 1.0, 2.0));
+  inMesh->createVertex(Eigen::Vector3d(0.0, 1.0, 4.0));
+  inMesh->createVertex(Eigen::Vector3d(0.0, 2.0, 5.0));
+  inMesh->createVertex(Eigen::Vector3d(0.0, 1.0, 7.0));
+  inMesh->allocateDataValues();
+  Eigen::VectorXd &inValues = inData->values();
+  inValues << 1.0, 3.0, 5.0, 7.0, 2.0, 4.0;
+
+  // Create mesh to map to
+  PtrMesh outMesh(new Mesh("OutMesh", dimensions, testing::nextMeshID()));
+  PtrData outData   = outMesh->createData("OutData", 1, 2_dataID);
+  int     outDataID = outData->getID();
+  outMesh->createVertex(Eigen::Vector3d::Constant(0.0));
+  outMesh->createVertex(Eigen::Vector3d(0.0, 0.0, 3.0));
+  outMesh->createVertex(Eigen::Vector3d(0.0, 0.0, 6.0));
+  outMesh->allocateDataValues();
+
+  // Setup mapping with mapping coordinates and geometry used
+  precice::mapping::RadialGeoMultiscaleMapping mapping(mapping::Mapping::CONSISTENT, dimensions, mapping::RadialGeoMultiscaleMapping::COLLECT, mapping::RadialGeoMultiscaleMapping::Z);
+  mapping.setMeshes(inMesh, outMesh);
+  BOOST_TEST(mapping.hasComputedMapping() == false);
+
+  // Map data
+  mapping.computeMapping();
+  mapping.map(inDataID, outDataID);
+  const Eigen::VectorXd &outValues = outData->values();
+
+  // Check if data is mapped to closest vertex
+  BOOST_TEST(mapping.hasComputedMapping() == true);
+  BOOST_TEST(outValues(0) == (inValues(0) + inValues(1)) / 2);
+  BOOST_TEST(outValues(1) == (inValues(2) + inValues(3)) / 2);
+  BOOST_TEST(outValues(2) == (inValues(4) + inValues(5)) / 2);
```

Comment:
I am not sure I understand this part.
The mapping should be similar to nearest-neighbor, not nearest-projection.
Why should the value of a node of `outMesh` be the average of two values of the `inMesh`?

If these are just expected due to the value choices in the test, then hard-coded values would make more sense here as well.

Reply:
From my understanding, the 1D vertex value should be the average of all 3D vertex values in it's vicinity. By vicinity I mean each 3D vertex that would be assigned the 1D value, if we did a "spread" mapping. If that is incorrect I need to change the mapping, maybe we can talk about that again.

Reply:
As discussed, we will write a comment about all mappings (Axial::Spread, Axial::Collect, Radial::Spread, Radial::Collect) and theit inner workings for the website documentation.